app/src/main/java/protect/card_locker/DBHelper.java

```diff
@@ -144,11 +144,17 @@ public boolean deleteLoyaltyCard (final int id)
         return (rowsDeleted == 1);
     }
 
-    public Cursor getLoyaltyCardCursor()
+    public Cursor getLoyaltyCardCursor(final String filter)
```

Comment:
As this call is used in two different contexts (with and without a filter), could a new get be added instead that supports the filter? e.g.

public Cursor getLoyaltyCardCursor() 
{
   // as normal
}

/**
 * Some javadoc that explains this call and what filter does
 */
public Cursor getLoyaltyCardCursor(final String filter) 
{
    // the new call that also can filter
}


Additionally, can there be a unit test that covers this new functionality? (Or are there new tests but I missed them?)

Reply:
Oh, I forgot Java can have the same function defined multiple times with different parameters (I normally develop Python). Yeah, that would be much cleaner indeed! Will do!

I haven't added any tests, but I'll try to add some.